src/modules/roc_fec/reader.cpp

```diff
@@ -552,19 +429,87 @@ void Reader::update_repair_block_() {
     }
 }
 
+bool Reader::process_source_packet_(const packet::PacketPtr& pp) {
+    const packet::FEC& fec = *pp->fec();
+
+    if (!validate_incoming_source_packet_(pp)) {
+        return false;
+    }
+
+    if (!check_payload_size_(fec.payload.size())) {
+        return false;
+    }
+
+    if (!check_source_block_size_(fec.source_block_length)) {
+        return false;
+    }
+
+    if (!update_payload_size_(fec.payload.size())) {
```

Comment:
It is still possible to update payload size and fail to update source block size.

Reply:
If update_source_block_size_() or update_repair_block_size_() fails, alive_ is always set to false, so that's ok.